Added password cracker lore (#30)
illinois-cs241/assignment-docs: password_cracker.md

  - "Synchronization, Part 5: Condition Variables"
  - "Synchronization, Part 6: Implementing a barrier"
---
## Backstory

> After finally being taken off of PIP due to your outstanding malloc performance you can finally relax a little. So take a Pomelo ElCroix out of the minifridge and relax!

Reviewer: ```suggestion
After finally being taken off of PIP due to your outstanding malloc performance, you can finally relax a little. So you take a Pomelo ElCroix out of the mini-fridge and relax!
```

Author: Tonally I felt it was better to exclude the "you" so it came across as more of an encouragement and less as a narration

Reviewer: Sounds good!

> Don't lower your standing desk just yet though because as it turns out, while you were competing an intern deleted the plaintext file containing everyone's passwords and now

Reviewer: ```suggestion
Don't lower your standing desk just yet, because as it turns out, while you were competing, an intern deleted the plaintext file containing everyone's passwords.
```

> nobody can log in. All that's left are the hashes in the auth database. Before you can update your LoopedIn status, your manager sprints over to your desk and tells you

Reviewer: ```suggestion
Now, nobody can log in. All that's left are the hashes in the auth database. Before you can update your LoopedIn status, your manager sprints over to your desk and tells you
```

that the company has freed up some resources and they want you to brute force everyone's passwords before they can take their PTO. Your manager suggests you use a thread

> pool but before you can read the NerdsForNerds article on what a thread pool is your manager's manager sends you a calendar invite telling you to parallelize each password.

Reviewer: ```suggestion
pool but before you can read the NerdsForNerds article on what a thread pool is, your manager's manager sends you a calendar invite telling you to parallelize the password cracking.
```

> Before you can accept without response your manager tells you to just do it both ways! It seems like a tough task but after conquering malloc nothing should be too hard right?

Reviewer: ```suggestion
Before you can accept without response, your manager tells you to just do it both ways! It seems like a tough task but after conquering `malloc`, nothing should be too hard right?
```

Reviewer: Also what do you mean by "accept without response"?

Author: In Outlook when you accept a calendar invite you can either accept normally or accept without response
